devito/dimension.py

```diff
@@ -367,20 +371,6 @@ def condition(self):
     def _hashable_content(self):
         return (self.parent._hashable_content(), self.factor, self.condition)
 
-    def _arg_defaults(self, **kwargs):
-        """
-        A :class:`ConditionalDimension` provides no arguments, so this
-        method returns an empty dict.
-        """
-        return {}
-
-    def _arg_values(self, *args, **kwargs):
```

Comment:
Can you update the docstring of ConditionalDimension explaining why it might need runtime arguments?